csharp/src/Ice/Communicator.cs

```diff
@@ -360,7 +373,126 @@ private class ObserverUpdater : Instrumentation.IObserverUpdater
 
                 TraceLevels = new TraceLevels(this);
 
-                DefaultsAndOverrides = new DefaultsAndOverrides(this, Logger);
+                DefaultHost = GetProperty("Ice.Default.Host");
+
+                if (GetProperty("Ice.Default.SourceAddress") is string address)
+                {
+                    DefaultSourceAddress = Network.GetNumericAddress(address);
+                    if (DefaultSourceAddress == null)
+                    {
+                        throw new InvalidConfigurationException(
+                            $"invalid IP address set for Ice.Default.SourceAddress: `{address}'");
+                    }
+                }
+
+                DefaultTransport = GetProperty("Ice.Default.Transport") ?? "tcp";
+
+                DefaultCollocationOptimization = (GetPropertyAsInt("Ice.Default.CollocationOptimized") ?? 1) > 0;
+
+                string val = GetProperty("Ice.Default.EndpointSelection") ?? "Random";
```

Comment:
I think I will prefer switch/case here, but feel free to keep it as is

Reply:
switches seem a little overkill to me for just 2 cases, so I'm probably going to keep it. Of course you can always silently change it in your future cleanup PRs 😄

Reply:
maybe a switch expression 
```
var endpointSelection = GetProperty("Ice.Default.EndpointSelection") ?? "Random";
DefaultEndpointSelection = endpointSelection switch
{
    "Random" => EndpointSelectionType.Random,
    "Ordered" => EndpointSelectionType.Ordered,
    _ => throw new InvalidConfigurationException(
        $"illegal value `{endpointSelection}'; expected `Random' or `Ordered'")
};```